library-best-practices/JLBP-17.md

```diff
@@ -0,0 +1,62 @@
+[JLBP-17] Coordinate Major Version Adoption
+-------------------------------------------
+
+For any particular library dependency chain, the libraries involved need to
+coordinate the adoption of a new major version of any particular shared library
+which doesn't rename its package so that consumers can continue using all
+functionality together. (This best practice doesn't apply to adoption of new
+major versions where the java package was renamed, because the old and new
+packages can co-exist.)
+
+As an example of what happens when this is not followed, imagine a partial
+adoption of a new major version of a leaf library. Start with the following
+dependency chain (both gax-java and grpc-java depend on guava 20.0):
+
+```
+gax-java 1.29.0 ->
+  grpc-java 1.13.1 ->
+    guava 20.0
+  guava 20.0
+```
+
+Next, guava releases a later version (26.0) which has a breaking change (a
+method is deleted). If gax-java 1.29.0 and grpc-java 1.13.1 depend on the
```

Comment:
"If gax-java 1.29.0 and grpc-java 1.13.1 depend on the
feature that is deleted, then it would be impossible" --> "If gax-java 1.29.0 and grpc-java 1.13.1 invoke that method, it is impossible"

Reply:
Will do.